server/src/main/java/org/candlepin/model/EntitlementCurator.java

```diff
@@ -364,6 +481,33 @@ public Entitlement replicate(Entitlement ent) {
         return criteria.list();
     }
 
+    /**
+     * Batch deletes a list of entitlements.
+     * @param pools
+     */
+    public void batchDelete(List<Entitlement> entitlements) {
+        for (Entitlement ent : entitlements) {
+            log.debug("Deleting entitlement: " + ent);
+            log.debug("certs.size = " + ent.getCertificates().size());
+
+            for (EntitlementCertificate cert : ent.getCertificates()) {
+                getEntityManager().remove(cert);
+            }
+            ent.getCertificates().clear();
+            getEntityManager().remove(ent);
+        }
+
+        // Maintain runtime consistency.
+        for (Entitlement ent : entitlements) {
+            ent.getCertificates().clear();
+            // Pool.entitlements is EXTRA Lazy. Before removing anything
+            // from that collection, its necessary to initialize it
+            Hibernate.initialize(ent.getPool().getEntitlements());
```

Comment:
I think EXTRA lazy is just buggy when combined with CascadeType.CREATE (yes CREATE, its not a typo :-) 

I created a unit test that shows this weird behavior (EntitlementCuratorTest.removeConsumerWithEntitlements) you can run it and debug it yourself. Without calling Hibernate.initialize it will fail with nasty SQL exceptions. The reason is, I think, that the if you remove the entitlement from the extra lazy collection, Hibernate will create it again due to CREATE cascading.